app/models/user.rb

```diff
@@ -13,9 +13,34 @@ def games
     Game.where("player_1_id = ? OR player_2_id = ?", id, id)
   end
 
-  def rival
-    # find who you've lost to the most
+  def rivals
     player_games = games
-    player_games.where("loser = ?", self)
+    lost_games = []
+    rivals = {}
+    player_games.each do |game|
+      if game.loser == self
+        lost_games << game
+      end
+    end
+    lost_games.each do |game|
+      if rivals.has_key?(game.winner.username)
+        rivals[game.winner.username] += 1
+      else
+        rivals[game.winner.username] = 1
+      end
+    end
+    if rivals == {}
+      rivals = {"n/a" => 0}
+    else
+      rivals = rivals.sort{|a,b| b[1] <=> a[1]}
```

Comment:
Space missing after comma.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.